PatientWebApplication/HealthClinic.DAL/Service/DoctorService.cs

```diff
@@ -224,33 +226,56 @@ public bool checkIfDoctorIsBusyForOperation(Operation operation, TimeSpan select
             return false;
         }
 
-        public Boolean doesDoctorHaveAnOperationAtSpecificTime(DoctorUser doctor, TimeSpan time, string date)
+        public Boolean DoesDoctorHaveAnOperationAtSpecificTime(DoctorUser doctor, TimeSpan time, string date)
         {
-            List<Operation> listOfOperation = operationRepository.GetAll();
-            if (listOfOperation == null) listOfOperation = new List<Operation>();
+            List<Operation> listOfOperation = _operationRepository.GetOperationsForDoctor(doctor.id);
+            if (listOfOperation == null) return false;
             foreach (Operation operation in listOfOperation)
             {
-                DoctorUser doctorOnOperation = operation.Doctor;
-                if (isDoctorsEquals(doctorOnOperation, doctor) && areDatesEqual(operation.Date,date) && checkIfDoctorIsBusyForOperation(operation, time))
+                if (areDatesEqual(operation.Date, date) && checkIfDoctorIsBusyForOperation(operation, time))
                 {
-                    return true; 
+                    return true;
                 }
             }
             return false;
         }
 
         public bool doesDoctorHaveAnOperationAtSpecificPeriod(DoctorUser doctor, TimeSpan start, TimeSpan end, string date)
         {
-            List<Operation> listOfOperation = operationRepository.GetAll();
+            List<Operation> listOfOperation = _operationRepository.GetAll();
             foreach (Operation operation in listOfOperation)
             {
                 DoctorUser doctorOnOperation = operation.Doctor;
                 if (isDoctorsEquals(doctorOnOperation, doctor) && areDatesEqual(operation.Date, date) && (checkIfDoctorIsBusyForOperation(operation, start) || checkIfDoctorIsBusyForOperation(operation, end)))
                     return true;
-                }
+            }
             return false;
         }
 
+        private List<DoctorUser> GetDoctorsBySpecialty(string specialty)
+        {
+            List<DoctorUser> specialists = new List<DoctorUser>();
+            foreach (DoctorUser doctor in GetAll())
+            {
+                if (UtilityMethods.CheckForSpecialty(doctor, specialty))
+                {
+                    specialists.Add(doctor);
+                }
+            }
+            return specialists;
```

Comment:
Refactor code. Replace method GetDoctorsBySpecialty(string specialty)  with Where syntax ,for example : 
return GetAll().Where(doctor => UtilityMethods.CheckForSpecialty(doctor, specialty) )